src/storage/predicate/column_predicate.cpp

```diff
@@ -33,21 +34,38 @@ static bool isColumnRef(ExpressionType type) {
     return type == ExpressionType::PROPERTY || type == ExpressionType::VARIABLE;
 }
 
+static bool isCastedColumnRef(const Expression& expr) {
+    if (expr.expressionType == ExpressionType::FUNCTION) {
+        const auto& funcExpr = expr.constCast<ScalarFunctionExpression>();
+        if (funcExpr.getFunction().name.starts_with("CAST") && funcExpr.getNumChildren() > 0) {
```

Comment:
Isn't checking function name starts_with "CAST" guarantees numChildren > 0? I think the alter part should be an KU_ASSERT.

Reply:
changed to assert